### ISSUE TITLE
Fix PublishContentController.Search method to ensure it does proper paging with Examine

### ISSUE THREAD

**Issue by Shazwazza** [umbraco/UmbracoRestApi#25]
Currently the paging is done in post processing with Linq which is horribly inneficient.

We need to fix the TODO in the code

````
//TODO: This would be more efficient if we went straight to the ExamineManager and used it's built in Skip method
// but then we have to write our own model mappers and don't have time for that right now.
```

**Comment by perploug** [comment]
Yes we indeed do! - one thing to be aware of though - we must ensure that we return proper IPublished content - and not a mapped Examine Record. 

Otherwise data between the 2 will be different, as a lucene record does not contain the same rich data as a published piece of content would - for instance a content picker would in a published state return a json blob - from lucene it would be 1234,1244,1234 format

at least that is how I understand examine to work - I might be wrong :)

**Comment by Shazwazza** [comment]
kind of sort of i don't think we have a proper examine to IPublishedContent mapper in the core yet either. We'll have to resort to searching via examine and then looking up each item in cache by Id which isn't very nice but that's what we'll do for now.
